ArmPkg/Drivers/SmmuDxe/README.md

```diff
@@ -244,27 +244,21 @@ Generic Platform Integration:
 - Append the IORT structure to this struct and update the fields accordingly.
 
   ```c
-   // SMMU_STATUS structure to hold the status of each SMMU instance.
-   // Contains the base address of the SMMU and whether it is enabled or not.
-   typedef struct _SMMU_STATUS {
-      UINT64     SmmuBase;
-      BOOLEAN    Enabled;
-   } SMMU_STATUS;
-
    // SMMU_CONFIG structure to pass the SMMU configuration data from the platform to the SMMU driver.
    typedef struct _SMMU_CONFIG {
-      UINT32         VersionMajor;
-      UINT32         VersionMinor;
-      UINT32         SmmuCount;
-      SMMU_STATUS    *SmmuStatus;
-      UINT32         IortSize;
-      UINT32         IortOffset;
+      UINT32    VersionMajor;
+      UINT32    VersionMinor;
+      UINT64    SmmuDisabledCount;                // Number of SMMU instances to be disabled.
+      UINT64    SmmuDisabledList[MAX_SMMU_COUNT]; // Array of disabled SMMU stream IDs.

```

Comment:
Consider using the offset scheme here as well. It's pretty typical to have a header appended with variable size arrays found by offsets and counts. This way you dont have to always have MAX_SMMU_COUNT as wasted space and a limitation.